File: chroma-league/src/main/java/net/booone/chromaleague/hud/parts/dragons/ElderBuffAnimation.java
package net.booone.chromaleague.hud.parts.dragons;

import net.booone.razersdk.animation.AnimatedFrame;
import net.booone.razersdk.animation.Frame;
import net.booone.razersdk.animation.SimpleFrame;
import net.booone.razersdk.color.StaticColor;
import net.booone.razersdk.sdk.RzKeyJoiner;

import java.util.List;

import static net.booone.razersdk.sdk.RzKey.*;

public class ElderBuffAnimation extends AnimatedFrame {

    @Override
    public Frame getFrame() {
        if (!hasFrame()) {
            extendAnimation();
        }
        return super.getFrame();
    }

    private void extendAnimation() {
        addAnimationFrame(2, new SimpleFrame(new RzKeyJoiner()
                .with(KilledDragonsBar.FOURTH_DRAGON_ROW).with(RZKEY_RALT)
                .with(KilledDragonsBar.SOUL_BAR)
                .with(RZKEY_LEFT, RZKEY_DOWN, RZKEY_RIGHT).join(), StaticColor.WHITE));
        addAnimationFrame(2, new SimpleFrame(List.of(RZKEY_RSHIFT,
                RZKEY_UP, RZKEY_SLASH), StaticColor.WHITE));
        addAnimationFrame(2, new SimpleFrame(List.of(RZKEY_ENTER, RZKEY_APOSTROPHE), StaticColor.WHITE));
        addAnimationFrame(2, new SimpleFrame(List.of(RZKEY_BACKSLASH, RZKEY_SQUARE_BRACKET_RIGHT), StaticColor.WHITE));
        addAnimationFrame(20, new SimpleFrame());
    }
}

File: src/hud/parts/dragons/ElderBuffAnimation.ts
import {AnimatedFrame} from '../../../razer-sdk/animation/AnimatedFrame.js';
import {Frame} from '../../../razer-sdk/animation/Frame.js';
import {SimpleFrame} from '../../../razer-sdk/animation/SimpleFrame.js';
import {StaticColor} from '../../../razer-sdk/color/StaticColor.js';
import {RzKey} from '../../../razer-sdk/sdk/RzKey.js';
import {RzKeyJoiner} from '../../../razer-sdk/sdk/RzKeyJoiner.js';
import {FOURTH_DRAGON_ROW, SOUL_BAR} from './KilledDragonsBar.js';

export class ElderBuffAnimation extends AnimatedFrame {
    override pollFrame(): Frame {
        if (!this.hasFrame()) {
            this.extendAnimation();
        }
        return super.pollFrame();
    }

    private extendAnimation(): void {
        this.addAnimationFrame(2, new SimpleFrame(
            [...new RzKeyJoiner()
                .withCollection(FOURTH_DRAGON_ROW)
                .withKeys(RzKey.RZKEY_RALT)
                .withCollection(SOUL_BAR)
                .withKeys(RzKey.RZKEY_LEFT, RzKey.RZKEY_DOWN, RzKey.RZKEY_RIGHT)
                .join()],
            StaticColor.WHITE,
        ));
        this.addAnimationFrame(2, new SimpleFrame(
            [RzKey.RZKEY_RSHIFT, RzKey.RZKEY_UP, RzKey.RZKEY_SLASH],
            StaticColor.WHITE,
        ));
        this.addAnimationFrame(2, new SimpleFrame(
            [RzKey.RZKEY_ENTER, RzKey.RZKEY_APOSTROPHE],
            StaticColor.WHITE,
        ));
        this.addAnimationFrame(2, new SimpleFrame(
            [RzKey.RZKEY_BACKSLASH, RzKey.RZKEY_SQUARE_BRACKET_RIGHT],
            StaticColor.WHITE,
        ));
        this.addAnimationFrame(20, new SimpleFrame());
    }
}
